.github/workflows/release.yaml

```diff
@@ -37,3 +37,4 @@ jobs:
         with:
           name: Release ${{ steps.get_version.outputs.VERSION }}
           files: splunk-otel-javaagent-all.jar
+          draft: true
```

Comment:
The release job should create draft release by default - we will hit the "make it official" button manually once we fill in the release notes.